synapse/config/experimental.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from synapse.config._base import Config
+from synapse.types import JsonDict
+
+
+class ExperimentalConfig(Config):
+    """Config section for enabling experimental features"""
+
+    section = "experimental"
+
+    # MSC2858 (multiple SSO identity providers)
+    msc2858_enabled = False
+
+    def read_config(self, config: JsonDict, **kwargs):
+        experimental = config.get("experimental_features")
+        if not experimental:
+            return
```

Comment:
Can we not just do `experimental = config.get("experimental_features") or {}`, rather than returning here? I find it a bit confusing to define `msc2858_enabled` in multiple places (with potentially different defaults)

Reply:
yes, fair